src/parser/warrior/fury/modules/talents/FrothingBerserker.js

```diff
@@ -0,0 +1,108 @@
+import React from 'react';
+
+import Analyzer from 'parser/core/Analyzer';
+
+import SPELLS from 'common/SPELLS/index';
+import SpellIcon from 'common/SpellIcon';
+import SpellLink from 'common/SpellLink';
+import { formatPercentage } from 'common/format';
+import StatisticBox from 'interface/others/StatisticBox';
+
+/**
+ Frothing Berserker users should maximize casts with 100 rage to proc Frothing Berserker the post
+ Carnage users should cast as soon as possible, i.e. at 70 rage
+ Massacre users (not from Soul of the Battlelord) should cast as soon as possible, i.e. at 85 rage
+ Whenever Battle Cry is active, user should cast Rampage whenever possible - regardless of talents
+ */
+
+class FrothingBerserker extends Analyzer {
+
+  last_timestamp = null;
+  casts_counter = 0;
+  premature_counter = 0;
+  reckless_abandon = null;
+
+  constructor(...args) {
+    super(...args);
+    this.active = this.selectedCombatant.hasTalent(SPELLS.FROTHING_BERSERKER_TALENT.id);
+    this.reckless_abandon = this.selectedCombatant.hasTalent(SPELLS.RECKLESS_ABANDON_TALENT.id);
+  }
+
+  on_byPlayer_cast(event) {
+    if (event.ability.guid === SPELLS.RAMPAGE.id) {
+      const rage = Math.floor(event.classResources[0].amount / 10);
+
+      // Ignore any free casts due to Massacre talent
+      if (this.selectedCombatant.hasBuff(SPELLS.MASSACRE.id)) {
+        return;
+      }
+
+      this.casts_counter++;
+
+      if (rage < 100 && !this.selectedCombatant.hasBuff(SPELLS.RECKLESSNESS.id)) {
+        this.premature_counter++;
+        this.last_timestamp = event.timestamp;
+      }
+    } else if (event.ability.guid === SPELLS.RECKLESSNESS.id && this.last_timestamp && this.reckless_abandon) {
+      if (event.timestamp - this.last_timestamp < 2000) {
```

Comment:
Avoid using magic numbers like 2000, if you have a cast buffer or something like it, it's good to create a const above the class with a name like PREMATURE_RECKLESSNESS_WINDOW or the like.